build_tools/update_environments_and_lock_files.py

```diff
@@ -225,11 +225,11 @@ def remove_from(alist, to_remove):
         "pip_dependencies": (
             remove_from(common_dependencies, ["python", "blas", "pip"])
             + docstring_test_dependencies
+            # Test with some optional dependencies
             + ["lightgbm", "scikit-image"]
+            # Test array API on CPU without PyTorch
+            + ["array-api-compat", "array-api-strict"]
         ),
-        "package_constraints": {
-            "python": "3.9",
-        },
```

Comment:
I removed this constraint because this build config is named "pylatest_*" and there was no inline comment to justify the Python version pin.